java/src/main/java/ai/rapids/cudf/HostMemoryBuffer.java

```diff
@@ -94,6 +117,23 @@ private HostMemoryBuffer(long address, long lengthInBytes, HostMemoryBuffer pare
     super(address, lengthInBytes, parent);
   }
 
+  /**
+   * Return a ByteBuffer that provides access to the underlying memory.  Please note <ol></ol> if the buffer is
+   * larger than a ByteBuffer can handle (2GB) only the first part of it will be exposed.  Also be
+   * aware that the ByteBuffer will be in native endian order, which is different from
+   */
+  public final ByteBuffer asByteBuffer() {
+    long len = Math.min(Integer.MAX_VALUE, length);
+    return asByteBuffer(0, len);
+  }
+
+  public final ByteBuffer asByteBuffer(long offset, long length) {
```

Comment:
Should this just take an `int` length to be more explicit about the limitation?  Then the compiler can help out.

Reply:
I did it because the C API exposes a long, but looking at the java API it is an int for allocation.